packages/xstate-wallet/src/workflows/create-and-fund.ts

```diff
@@ -97,11 +97,28 @@ const triggerObjective = (store: Store) => async (ctx: Init): Promise<void> => {
 const assignJointChannelId = assign<VirtualFundingComplete>({
   jointChannelId: (_, event: DoneInvokeEvent<{jointChannelId: string}>) => event.data.jointChannelId
 });
+
+const reserveFunds = (store: Store) => async (context, event) => {
+  const channelEntry = await store.getEntry(context.channelId);
+  const {allocationItems} = checkThat(channelEntry.supported.outcome, isSimpleEthAllocation);
+  const playerAddress = await store.getAddress();
+  const playerDestination =
+    channelEntry.supported.participants.find(p => p.signingAddress === playerAddress)
```

Comment:
This repeated pattern makes me think that `entry.participants` can be a class that exposes a `.withParticipantId(id: string)` to find the given participant, or error out if it doesn't exist.